src/test/unit/regions/VectorFileTest.cpp

```diff
@@ -0,0 +1,327 @@
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ 
+ * Author: David Keeney, April, 2019
+ * ---------------------------------------------------------------------
+ */
+ 
+/*---------------------------------------------------------------------
+  * This is a test of the VectorFileEffector and VectorFileSensor modules.  
+  *
+  * For those not familiar with GTest:
+  *     ASSERT_TRUE(value)   -- Fatal assertion that the value is true.  Test terminates if false.
+  *     ASSERT_FALSE(value)   -- Fatal assertion that the value is false. Test terminates if true.
+  *     ASSERT_STREQ(str1, str2)   -- Fatal assertion that the strings are equal. Test terminates if false.
+  *
+  *     EXPECT_TRUE(value)   -- Nonfatal assertion that the value is true.  Test fails but continues if false.
+  *     EXPECT_FALSE(value)   -- Nonfatal assertion that the value is false. Test fails but continues if true.
+  *     EXPECT_STREQ(str1, str2)   -- Nonfatal assertion that the strings are equal. Test fails but continues if false.
+  *
+  *     EXPECT_THROW(statement, exception_type) -- nonfatal exception, cought and continues.
+  *---------------------------------------------------------------------
+  */
+
+
+#include <nupic/engine/NuPIC.hpp>
+#include <nupic/engine/Network.hpp>
+#include <nupic/engine/Region.hpp>
+#include <nupic/engine/Spec.hpp>
+#include <nupic/engine/Input.hpp>
+#include <nupic/engine/Output.hpp>
+#include <nupic/engine/Link.hpp>
+#include <nupic/engine/RegisteredRegionImpl.hpp>
+#include <nupic/engine/RegisteredRegionImplCpp.hpp>
+#include <nupic/ntypes/Array.hpp>
+#include <nupic/types/Exception.hpp>
+#include <nupic/os/Env.hpp>
+#include <nupic/os/Path.hpp>
+#include <nupic/os/Timer.hpp>
+#include <nupic/os/Directory.hpp>
+#include <nupic/engine/YAMLUtils.hpp>
+#include <nupic/regions/SPRegion.hpp>
+#include <nupic/utils/LogItem.hpp>
+
+
+#include <string>
+#include <vector>
+#include <cmath> // fabs/abs
+#include <cstdlib> // exit
+#include <iostream>
+#include <stdexcept>
+#include <fstream>
+#include <streambuf>
+#include <iterator>
+#include <algorithm>
+
+
+
+#include "yaml-cpp/yaml.h"
+#include "gtest/gtest.h"
+#include "RegionTestUtilities.hpp"
+
+#define VERBOSE if(verbose)std::cerr << "[          ] "
```

Comment:
wasn't this macro DEBUG in other files?

Reply:
> wasn't this macro DEBUG in other files?

I have always used VERBOSE.  NTA_DEBUG is a very different thing, see Log.hpp.  DEBUG is used in SlidingWindow.hpp as a variable...(not a good idea because the compiler may have a different meaning.)